core/src/main/java/io/questdb/cairo/wal/WalWriter.java

```diff
@@ -347,6 +347,12 @@ public boolean isOpen() {
         return this.open;
     }
 
+    @Override
+    public boolean isPartitionReadOnlyByTimestamp(long timestamp) {
+        // wall does not know whether a partition is read only at present 
+        return false;
```

Comment:
Please throw error here to not hide the problem

Reply:
Is there way to not have this method on `TableWriterAPI`? We’re moving towards WAL being main table type and it’s not possible to have partition information for WAL at the moment of writing

Reply:
there is a way, but I am not sure of its correctness, I have implemented it anyway. So now we do not check at the point of Line event appending, instead we create the Line event..... and let o3Commit deal with the partition read-only situation. Here is where I think it is tricky: if you try to insert something in a read-only partition..... no error message, just a silent ignore. In a way I think it is better because no legit data are loss, but I fear users might raise an eyebrow.

Reply:
so instead this will be logged at critical level, so users become aware.